lightning/src/chain/keysinterface.rs

```diff
@@ -833,9 +865,14 @@ impl KeysManager {
 				};
 				let channel_master_key = master_key.ckd_priv(&secp_ctx, ChildNumber::from_hardened_idx(3).unwrap()).expect("Your RNG is busted");
 				let rand_bytes_master_key = master_key.ckd_priv(&secp_ctx, ChildNumber::from_hardened_idx(4).unwrap()).expect("Your RNG is busted");
-				let inbound_payment_key: SecretKey = master_key.ckd_priv(&secp_ctx, ChildNumber::from_hardened_idx(5).unwrap()).expect("Your RNG is busted").private_key.key;
-				let mut inbound_pmt_key_bytes = [0; 32];
-				inbound_pmt_key_bytes.copy_from_slice(&inbound_payment_key[..]);
+				let inbound_payment_key = if let Some(key) = inbound_payment_key_opt {
+					key
```

Comment:
bleh, quite the layer violation - if we're returning the key material  to `ChannelManager` as-is and letting `ChannelManager` do derivation magic on it, we should definitely not be doing some derivation on it ourselves. Instead, lets pass in a cross_node_seed or something that is then derived into both a `inbound_payment_key` and a `phantom_secret` directly. This avoids any (ok, admittedly far-fetched, but still) use of the same derivation in `ChannelManager` leading to key re-use (eg if `ChannelManager` decides to derive a key called "LDK Phandom Node Secret Key Expansion).

Reply:
Hmm, seems like it may be awkward to change the derivation of the `inbound_payment_key` given it's already used in the wild. Is it fine if some payments fail due to this change?

Reply:
Ah, right, hmmmmm, no that wouldn't be great. Maybe we should separate out how it works based on the constructor used? For the `new` constructor we can just refuse to generate phantom secret?

Reply:
Implemented this